src/pages/UserProfile/UserProfile.jsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Flex, Stack } from '@chakra-ui/react';
+import Logout from '../../components/Logout/Logout';
+import UserDetails from '../../components/UserDetails/UserDetails';
+import styles from './UserProfile.module.css';
+
+const UserProfile = () => {
```

Comment:
use `useParams` to grab the userId from the url to pass into `UserDetails`. also enforce that if the user is a driver that their userId matches the url's userId, otherwise boot them back to `/login`